proto/substrait/algebra.proto

```diff
@@ -819,9 +819,12 @@ message Expression {
       UserDefined user_defined = 33;
     }
 
-    // whether the literal type should be treated as a nullable type. Applies to
-    // all members of union other than the Typed null (which should directly
-    // declare nullability).
+    // Whether the literal_type above should be treated as a nullable type.
+    // Applies to all members of the literal_type oneof EXCEPT:
+    //  * Type null
+    //  * Type.List empty_list
+    //  * Type.Map empty_map
+    // which should declare their nullability directly instead.
```

Comment:
I need to verify but the text format appears to set both for empty lists and just the type for empty maps (based on its test cases taken from Isthmus).

Reply:
@EpsilonPrime where you able to verify this?

Reply:
It shouldn't matter going forward, I will make sure that the text format follows this guideline by adding some tests.

Reply:
For empty lists the following behavior is present for the text format:

```
{}_list<string> -> nothing nullable
{}_list<string?> -> literal is nullable, list is not nullable
{}_list?<string> -> list is nullable, literal is not
{}_list?<string?> -> both are nullable
```

It seems like this nullability covers the nullability of the list even if it's empty.  Even if it's empty is it legal to have a nullable version of it?

Reply:
Added the empty_map tests in https://github.com/substrait-io/substrait-cpp/pull/97.

Reply:
The text format is definitely using this nullability over the one in the type.  It's the same behavior it uses for non-empty lists and maps.